ddtrace/contrib/wsgi/wsgi.py

```diff
@@ -80,73 +81,107 @@ def default_wsgi_span_modifier(span, environ):
     span.resource = "{} {}".format(environ["REQUEST_METHOD"], environ["PATH_INFO"])
 
 
-class DDWSGIMiddleware(object):
-    """WSGI middleware providing tracing around an application.
-
-    :param application: The WSGI application to apply the middleware to.
-    :param tracer: Tracer instance to use the middleware with. Defaults to the global tracer.
-    :param span_modifier: Span modifier that can add tags to the root span.
-                            Defaults to using the request method and url in the resource.
-    """
-
-    def __init__(self, application, tracer=None, span_modifier=default_wsgi_span_modifier):
+class DDWSGIMiddlewareBase(object):
+    def __init__(self, application, middleware_config, tracer):
```

Comment:
Shouldn't the default for `tracer` remain with a default?

```suggestion
    def __init__(self, application, middleware_config, tracer=None):
```

Reply:
I removed `ddtrace.tracer` from line 88 instead (this was a mistake). This is behavior I would like to move away from. The child class should always pass a tracer to DDWSGIMiddlewareBase.__init__()